cmd/reword/main.go

```diff
@@ -31,10 +34,42 @@ func main() {
 		}
 		writefile.Close()
 	} else {
-		file, err := os.Open(filename)
-		check(err)
-		file.Close()
+		if shouldAppendCommitID(filename) {
+			appendCommitID(filename)
+		}
+	}
+}
+
+func shouldAppendCommitID(filename string) bool {
+	readfile, err := os.Open(filename)
+	check(err)
+	defer readfile.Close()
+
+	i := 0
+	nonEmptyCommitMessage := false
+	scanner := bufio.NewScanner(readfile)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if line != "" && !strings.HasPrefix(line, "#") {
+			nonEmptyCommitMessage = true
+		}
+		if strings.HasPrefix(line, "commit-id:") {
+			return false
+		}
+		i++
 	}
+	check(scanner.Err())
+	return nonEmptyCommitMessage
+}
+
+func appendCommitID(filename string) {
+	appendfile, err := os.OpenFile(filename, os.O_APPEND|os.O_WRONLY, 0666)
+	check(err)
+	defer appendfile.Close()
+
+	commitID := uuid.New()
+	appendfile.WriteString("\n")
```

Comment:
If the user has only provided a commit title, then I think this would put the commit-id on the second line? By convention, the second line should be blank - https://blog.ssanj.net/posts/2015-10-22-git-commit-message-format.html.

Reply:
This is exactly the reason for this new line, to leave a blank line after the title or body. It's the same behavior as the current implementation using the commit-msg hook and has caused no issues that I'm aware of so far.

Reply:
I've found a case where this doesn't work. It seems like the regular Git CLI adds a newline after your commit message, and then spr adds another one so you get two lines.

When using [GitUp](https://gitup.co), it doesn't look like it adds the newlines so you get the commit-id rendered as part of the title:

* 0630d17122 - (HEAD -> master) This is my test commit commit-id:e78b859d (3 minutes ago) <Daniel Compton>

<img width="491" alt="Screenshot of GitUp (18-08-22, 10-16-27 AM)" src="https://user-images.githubusercontent.com/811954/185253376-1384934d-22dc-40a9-966b-6b17cd68b8c0.png">

![CleanShot 2022-08-18 at 10 16 37@2x](https://user-images.githubusercontent.com/811954/185253344-d1a3d5ce-e0a0-4eab-a009-00ab53c04f66.png)

<img width="360" alt="Screenshot of GitUp (18-08-22, 10-18-01 AM)" src="https://user-images.githubusercontent.com/811954/185253441-c8246008-a956-4616-8f3e-46f4b21458a2.png">

Reply:
what does `git log` output?
in my case with only a title I get:
```
commit 66c2d2ab49cfa2452cdcf98f9eae29d36d1d0f0e (HEAD -> master, origin/pr/ejoffe/master/dc5d2962)
Author: Eitan Joffe <ejoffe@gmail.com>
Date:   Wed Aug 17 14:23:01 2022 -0700

    title text

    commit-id:dc5d2962

commit 22d54c38f81b624d1bdc34352adb621df0e31d0d (origin/pr/ejoffe/master/0dc38ae2)
Author: Eitan Joffe <ejoffe@gmail.com>
```

Reply:
my guess is the gitup tool is using the commit-id as the "body" of the message (rightfully so).
you can try to rebase -i and add another newline between the title and the commit-id, see if that changes anything, but I suspect the outcome will be the same

Reply:
Here's the commit that's being created:

```console
$ git -P show
commit d4afe69d96535502b9eda518decd42ec9af655aa (HEAD -> dc/recents-navbar-stacked-pr)
Author: Daniel Compton <>
Date:   Thu Aug 18 11:07:33 2022 +1200

    abc
    commit-id:e78b859d

diff --git a/app/version.txt1 b/app/version.txt1
new file mode 100644
index 0000000000..e69de29bb2
```

```console
$ git log --graph --pretty=format:'%s %Cgreen(%cr)' --abbrev-commit --date=relative
* abc commit-id:e78b859d (3 minutes ago)
# ...
```

Adding an extra newline between the title and the body lets GitUp render the title correctly.

<img width="176" alt="Screenshot of GitUp (18-08-22, 11-13-01 AM)" src="https://user-images.githubusercontent.com/811954/185259521-7287485c-7d6d-4917-bf75-1f2f9eb75b2e.png">

Reply:
Alright, fix coming.